File: Branches/VBInicial/vpPriV100GrupoMundifios/Inovafil/Compras/EditorCompras/CmpIsEditorCompras.vb
Imports Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs
Imports Primavera.Extensibility.Purchases.Editors
Imports Primavera.Extensibility.Attributes
Imports Vimaponto.PrimaveraV100.Clientes.GrupoMundifios.Generico
Imports StdBE100
Imports BasBE100.BasBETiposGcp

Namespace Inovafil
    Public Class CmpIsEditorCompras
        Inherits EditorCompras

        Public Overrides Sub AntesDeGravar(ByRef Cancel As Boolean, e As ExtensibilityEventArgs)
            MyBase.AntesDeGravar(Cancel, e)

            If Module1.VerificaToken("Inovafil") = 1 Then



                '##############################################################################################################
                '#Envia e-mail para Joaquim António com Encomendas a Fornecedor que contenham o artigo Rama. JFC - 31/01/2019 #
                '##############################################################################################################
                If Me.DocumentoCompra.Tipodoc = "ECF" Then
                    Dim EnviaEmailRamaBolean As Boolean
                    Dim EnviaEmailCertificadoBolean As Boolean
                    EnviaEmailRamaBolean = False
                    EnviaEmailCertificadoBolean = False
                    Dim listValidaCabec As StdBELista

                    listValidaCabec = BSO.Consulta("select * from CabecCompras cc where cc.Id='" & Me.DocumentoCompra.ID & "'")

                    If listValidaCabec.Vazia = True Then
                        For i = 1 To Me.DocumentoCompra.Linhas.NumItens

                            If BSO.Base.Artigos.DaValorAtributo(Me.DocumentoCompra.Linhas.GetEdita(i).Artigo, "Descricao") Like "Rama*" Then

                                EnviaEmailRamaBolean = True

                            End If

                            If Me.DocumentoCompra.Linhas.GetEdita(i).CamposUtil("CDU_Artigo").Valor & "" = "" And Me.DocumentoCompra.Linhas.GetEdita(i).Artigo & "" <> "" Then
                                If UCase(Me.DocumentoCompra.Linhas.GetEdita(i).Descricao) Like "*RECYCLED*" Or UCase(Me.DocumentoCompra.Linhas.GetEdita(i).Descricao) Like "*RECICLADO*" Or UCase(Me.DocumentoCompra.Linhas.GetEdita(i).Descricao) Like "*REPREVE*" Or UCase(Me.DocumentoCompra.Linhas.GetEdita(i).Descricao) Like "*GRS*" Or UCase(Me.DocumentoCompra.Linhas.GetEdita(i).Descricao) Like "*GOTS*" Or UCase(Me.DocumentoCompra.Linhas.GetEdita(i).Descricao) Like "*OCS*" Or UCase(Me.DocumentoCompra.Linhas.GetEdita(i).Descricao) Like "*BCI*" Or UCase(Me.DocumentoCompra.Linhas.GetEdita(i).Descricao) Like "*SUPIMA*" Then


                                    EnviaEmailCertificadoBolean = True
                                End If
                            End If
                        Next i
                    End If


                    If EnviaEmailRamaBolean = True Then
                        EnvioEmailRama
                    End If
                    ''---Após alteração de funçoes da Filipa----Codigo comentado porque quem trata dos Certificados é a Ana e o Ricardo e são os mesmo que criam as ECF
                    '     If EnviaEmailCertificadoBolean = True Then
                    '    EnvioEmailCertificado
                    '    End If



                End If
                '##############################################################################################################
                '#Envia e-mail para Joaquim António com Encomendas a Fornecedor que contenham o artigo Rama. JFC - 31/01/2019 #
                '##############################################################################################################

            End If

        End Sub


        Public Overrides Sub ArtigoIdentificado(Artigo As String, NumLinha As Integer, ByRef Cancel As Boolean, e As ExtensibilityEventArgs)
            MyBase.ArtigoIdentificado(Artigo, NumLinha, Cancel, e)

            If Module1.VerificaToken("Inovafil") = 1 Then

                '#####################################################################################################################
                '####    Bruno 30/01/2020 Adiciona linha especial a pedido pela Andreia em casos de artigos Lã ou Cashemira  #########
                '#####################################################################################################################
                If Me.DocumentoCompra.Tipodoc = "ECF" Then


                    If Me.DocumentoCompra.Linhas.GetEdita(NumLinha).Descricao Like "*Lã*" Then
                        If Me.DocumentoCompra.Pais = "PT" Then
                            BSO.Compras.Documentos.AdicionaLinhaEspecial(Me.DocumentoCompra, vdTipoLinhaEspecial.vdLinha_Comentario, 0, "Artigo composto por Lã. Por favor envie o Relatório de Qualidade e a Declaração origem e de Non Mulesing.")
                        Else
                            BSO.Compras.Documentos.AdicionaLinhaEspecial(Me.DocumentoCompra, vdTipoLinhaEspecial.vdLinha_Comentario, 0, "This Product is composed by WO. Please send Quality Report and Declaration of Origin and Not Mulesing.")
                        End If
                    End If
                    If Me.DocumentoCompra.Linhas.GetEdita(NumLinha).Descricao Like "*Cashemira*" Then
                        If Me.DocumentoCompra.Pais = "PT" Then
                            BSO.Compras.Documentos.AdicionaLinhaEspecial(Me.DocumentoCompra, vdTipoLinhaEspecial.vdLinha_Comentario, 0, "Artigo composto por Cashemira. Por favor envie o Relatório de Qualidade e Certificado Laboratorial de composição.")
                        Else
                            BSO.Compras.Documentos.AdicionaLinhaEspecial(Me.DocumentoCompra, vdTipoLinhaEspecial.vdLinha_Comentario, 0, "This Product is composed by Cashemira. Please send Quality Report and Laboratory Certificate of composition")
                        End If
                    End If


                End If

                '#####################################################################################################################
                '####    Bruno 30/01/2020 Adiciona linha especial a pedido pela Andreia em casos de artigos Lã ou Cashemira  #########
                '#####################################################################################################################
            End If

        End Sub

        Public Overrides Sub DepoisDeGravar(Filial As String, Tipo As String, Serie As String, NumDoc As Integer, e As ExtensibilityEventArgs)
            MyBase.DepoisDeGravar(Filial, Tipo, Serie, NumDoc, e)

            If Module1.VerificaToken("Inovafil") Then

                '*******************************************************************************************************************************************
                '#### Atualizar o Armazem Stock Service na Mundifios 14/12/2018 - JFC
                '*******************************************************************************************************************************************
                Dim j As Long
                Dim AtualizaSS As Boolean
                AtualizaSS = False
                For j = 1 To Me.DocumentoCompra.Linhas.NumItens

                    If Me.DocumentoCompra.Linhas(j).Armazem = "INO" Then
                        AtualizaSS = True

                    End If

                Next j

                If AtualizaSS = True Then
                    BSO.DSO.ExecuteSQL("exec priinovafil.dbo.spAtualizaSS")
                End If

                '*******************************************************************************************************************************************
                '#### Atualizar o Armazem Stock Service na Mundifios 14/12/2018 - JFC
                '*******************************************************************************************************************************************
            End If
        End Sub


        Dim VarFrom As String
        Dim VarTo As String
        Dim VarAssunto As String
        Dim VarTextoInicialMsg As String
        Dim VarMensagem As String
        Dim VarUtilizador As String
        Dim VarLinhas As String

        Private Function EnvioEmailRama()



            Dim ln As Long
            VarFrom = ""

            VarTo = "informatica@mundifios.pt;jafernandes@mundifios.pt"

            If Format(Now, "HH:mm") >= "07:00" And Format(Now, "HH:mm") <= "12:59" Then
                VarTextoInicialMsg = "Bom dia,"
            ElseIf Format(Now, "HH:mm") >= "13:00" And Format(Now, "HH:mm") <= "19:59" Then
                VarTextoInicialMsg = "Boa tarde,"
            Else
                VarTextoInicialMsg = "Boa noite,"
            End If

            VarAssunto = "(Rama Inovafil) Encomenda a Fornecedor: " & Format(Me.DocumentoCompra.NumDoc, "####") & "/" & Me.DocumentoCompra.Serie

            VarUtilizador = Aplicacao.Utilizador.Utilizador

            VarLinhas = ""

            For ln = 1 To Me.DocumentoCompra.Linhas.NumItens

                VarLinhas = VarLinhas & "Linha " & ln & ":                         " & Me.DocumentoCompra.Linhas.GetEdita(ln).Artigo & " - Armazem:" & Me.DocumentoCompra.Linhas.GetEdita(ln).Armazem & " - Lote:" & Me.DocumentoCompra.Linhas.GetEdita(ln).Lote & " - Desc:" & Me.DocumentoCompra.Linhas.GetEdita(ln).Descricao & " - Quantidade:" & Me.DocumentoCompra.Linhas.GetEdita(ln).Quantidade & Me.DocumentoCompra.Linhas.GetEdita(ln).Unidade & " - Prec.Unit:" & Me.DocumentoCompra.Linhas.GetEdita(ln).PrecUnit & Me.DocumentoCompra.Moeda & " - Data Entrega:" & Me.DocumentoCompra.Linhas.GetEdita(ln).DataEntrega & Chr(13) & ""


            Next ln


            VarMensagem = VarTextoInicialMsg & Chr(13) & Chr(13) & Chr(13) & "Foi emitido uma Encomenda a Fornecedor no Primavera:" & Chr(13) & Chr(13) & "" _
                        & "Empresa:                         " & BSO.Contexto.CodEmp & " - " & BSO.Contexto.IDNome & Chr(13) & "" _
                        & "Utilizador:                      " & VarUtilizador & Chr(13) & Chr(13) & "" _
                        & "Fornecedor:                      " & Me.DocumentoCompra.Entidade & " - " & Replace(BSO.Base.Fornecedores.Edita(Me.DocumentoCompra.Entidade).Nome, "'", "") & Chr(13) & "" _
                        & "Documento:                       " & Me.DocumentoCompra.Tipodoc & " " & Format(Me.DocumentoCompra.NumDoc, "#,###") & "/" & Me.DocumentoCompra.Serie & Chr(13) & Chr(13) & Chr(13) & "" _
                        & VarLinhas & Chr(13) & "" _
                        & "Cumprimentos"




            BSO.DSO.ExecuteSQL("INSERT INTO [PRIEMPRE].[DBO].[MENSAGENSEMAIL]  ([Data], [From], [To], [CC], [BCC], [Assunto], [Mensagem], [Anexos], [Formato], [Utilizador]) VALUES('" & Format(Now, "yyyy-MM-dd HH:mm:ss") & "', '" & VarFrom & "', '" & VarTo & "','','','" & VarAssunto & "','" & VarMensagem & "','',0,'" & VarUtilizador & "' )")




        End Function

    End Class
End Namespace

File: Trunk/vpPriV100Inovafil/Inovafil/Compras/EditorCompras/CmpIsEditorCompras.cs
using Primavera.Extensibility.Purchases.Editors;
using Primavera.Extensibility.BusinessEntities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs;
using StdBE100;
using Microsoft.VisualBasic;
using static BasBE100.BasBETiposGcp;
using IntBE100;
using System.Windows.Forms;
using Generico;

namespace Inovafil
{
    public class CmpIsEditorCompras : EditorCompras
    {

        public override void AntesDeGravar(ref bool Cancel, ExtensibilityEventArgs e)
        {
            base.AntesDeGravar(ref Cancel, e);

            // ##############################################################################################################
            // #Envia e-mail para Joaquim António com Encomendas a Fornecedor que contenham o artigo Rama. JFC - 31/01/2019 #
            // ##############################################################################################################
            if (this.DocumentoCompra.Tipodoc == "ECF")
            {
                bool EnviaEmailRamaBolean;
                bool EnviaEmailCertificadoBolean;
                EnviaEmailRamaBolean = false;
                EnviaEmailCertificadoBolean = false;
                StdBELista listValidaCabec;

                listValidaCabec = BSO.Consulta("select * from CabecCompras cc where cc.Id='" + this.DocumentoCompra.ID + "'");


                if (listValidaCabec.Vazia() == true)
                {

                    for (int i = 1; i <= DocumentoCompra.Linhas.NumItens; i++)
                    {
                        string descricao = BSO.Base.Artigos.DaValorAtributo(DocumentoCompra.Linhas.GetEdita(i).Artigo, "Descricao");
                        if (descricao.Contains("Rama"))
                            EnviaEmailRamaBolean = true;

                        if (DocumentoCompra.Linhas.GetEdita(i).CamposUtil["CDU_Artigo"].Valor + "" == "" && DocumentoCompra.Linhas.GetEdita(i).Artigo + "" != "")
                        {

                            if (Strings.UCase(DocumentoCompra.Linhas.GetEdita(i).Descricao).Contains("RECYCLED") || Strings.UCase(DocumentoCompra.Linhas.GetEdita(i).Descricao).Contains("RECICLADO") || Strings.UCase(DocumentoCompra.Linhas.GetEdita(i).Descricao).Contains("REPREVE") || Strings.UCase(DocumentoCompra.Linhas.GetEdita(i).Descricao).Contains("GRS") || Strings.UCase(DocumentoCompra.Linhas.GetEdita(i).Descricao).Contains("GOTS") || Strings.UCase(DocumentoCompra.Linhas.GetEdita(i).Descricao).Contains("OCS") || Strings.UCase(DocumentoCompra.Linhas.GetEdita(i).Descricao).Contains("BCI") || Strings.UCase(DocumentoCompra.Linhas.GetEdita(i).Descricao).Contains("SUPIMA"))
                                EnviaEmailCertificadoBolean = true;


                        }

                    }
                }

                if (EnviaEmailRamaBolean == true)
                    EnvioEmailRama();


            }

        }


        public override void ArtigoIdentificado(string Artigo, int NumLinha, ref bool Cancel, ExtensibilityEventArgs e)
        {
            base.ArtigoIdentificado(Artigo, NumLinha, ref Cancel, e);

            if (DocumentoCompra.Tipodoc == "ECF")
            {

                if (DocumentoCompra.Linhas.GetEdita(NumLinha).Descricao.Contains("Lã"))
                {

                    if (DocumentoCompra.Pais == "PT")
                        BSO.Compras.Documentos.AdicionaLinhaEspecial(DocumentoCompra, compTipoLinhaEspecial.compLinha_Comentario, 0, "Artigo composto por Lã. Por favor envie o Relatório de Qualidade e a Declaração origem e de Non Mulesing.");
                    else
                        BSO.Compras.Documentos.AdicionaLinhaEspecial(DocumentoCompra, compTipoLinhaEspecial.compLinha_Comentario, 0, "This Product is composed by WO. Please send Quality Report and Declaration of Origin and Not Mulesing.");

                }

                if (DocumentoCompra.Linhas.GetEdita(NumLinha).Descricao.Contains("Cashemira"))
                {

                    if (DocumentoCompra.Pais == "PT")
                        BSO.Compras.Documentos.AdicionaLinhaEspecial(DocumentoCompra, compTipoLinhaEspecial.compLinha_Comentario, 0, "Artigo composto por Cashemira. Por favor envie o Relatório de Qualidade e Certificado Laboratorial de composição.");
                    else
                        BSO.Compras.Documentos.AdicionaLinhaEspecial(DocumentoCompra, compTipoLinhaEspecial.compLinha_Comentario, 0, "This Product is composed by Cashemira. Please send Quality Report and Laboratory Certificate of composition.");

                }

            }

        }


        public override void DepoisDeGravar(string Filial, string Tipo, string Serie, int NumDoc, ExtensibilityEventArgs e)
        {
            base.DepoisDeGravar(Filial, Tipo, Serie, NumDoc, e);


            for (int i = 1; i <= DocumentoCompra.Linhas.NumItens; i++)
            {
                if (DocumentoCompra.Linhas.GetEdita(i).Armazem == "INO")
                {
                    string query = " INSERT INTO PRIMUNDIFIOS.dbo.ArtigoLote(Artigo, Lote, Descricao, DataFabrico, Validade, Controlador, Activo, Observacoes, CDU_TipoQualidade, CDU_Parafinado)"
                    + " SELECT al.Artigo, "
                    + " al.Lote, "
                    + " al.Descricao, "
                    + " al.DataFabrico, "
                    + " al.Validade, "
                    + " al.Controlador, "
                    + " al.Activo, "
                    + " al.Observacoes, "
                    + " al.CDU_TipoQualidade, "
                    + " al.CDU_Parafinado "
                    + " FROM ArtigoLote al "
                    + " WHERE concat(al.Artigo, al.lote) in "
                    + " (SELECT concat(aa.Artigo, aa.Lote) "
                    + " FROM ArtigoArmazem aa "
                    + " WHERE aa.Armazem = 'INO' "
                    + " AND aa.Artigo in "
                    + " (SELECT a.Artigo "
                    + " FROM primundifios.dbo.Artigo a "
                    + " WHERE a.Artigo = aa.Artigo)) "
                    + " AND concat(al.Artigo, al.lote) not in "
                    + " (SELECT concat(al2.Artigo, al2.Lote) "
                    + " FROM primundifios.dbo.ArtigoLote al2 "
                    + " WHERE al2.Artigo = al.Artigo "
                    + " AND al2.Artigo IS NOT NULL "
                    + " AND al2.Lote IS NOT NULL) ";

                    BSO.DSO.ExecuteSQL(query);

                    IntBEDocumentoInterno DocInt = new IntBEDocumentoInterno();
                    StdBELista listaDoc = new StdBELista();

                    listaDoc = BSO.DSO.Consulta("select Numerador from PRIMUNDIFIOS.dbo.SeriesInternos where TipoDoc = 'ES' and Serie='INO'");
                    if (listaDoc.Valor("Numerador") == 0)
                    {
                        listaDoc = null;
                        listaDoc = BSO.DSO.Consulta("SELECT aa.Armazem,aa.Artigo, aa.Lote, aa.StkActual - isnull( (SELECT sum(ln.Quantidade-lns.QuantTrans) FROM CabecDoc cd INNER JOIN LinhasDoc ln ON ln.IdCabecDoc=cd.Id INNER JOIN CabecDocStatus cds ON cds.IdCabecDoc=cd.Id INNER JOIN LinhasDocStatus lns ON lns.IdLinhasDoc=ln.Id WHERE cds.Anulado='0' AND cd.TipoDoc='ECL' AND ln.Armazem=aa.Armazem AND ln.Artigo=aa.Artigo AND ln.Lote=aa.Lote AND lns.Fechado='0' AND lns.EstadoTrans='P' ),0) AS 'StkActual', aa.Localizacao, aa.PCMedio, aa.PCUltimo FROM ArtigoArmazem aa WHERE aa.Armazem='INO' AND aa.Artigo in (SELECT a.Artigo FROM primundifios.dbo.Artigo a WHERE a.Artigo=aa.Artigo) And aa.StkActual > 0");

                        Module1.AbreEmpresa("MUNDIFIOS");


                        DocInt.ID = PSO.FuncoesGlobais.CriaGuid(true);

                        DocInt.Tipodoc = "ES";
                        DocInt.Serie = "INO";

                        // Preenche a restante informação no documento
                        Module1.emp.Internos.Documentos.PreencheDadosRelacionados(DocInt);
                        // Data de Introdução
                        DocInt.Data = DateTime.Now;
                        DocInt.DataVencimento = DateTime.Now;
                    }
                    else
                    {
                        listaDoc = null;
                        listaDoc = BSO.DSO.Consulta("SELECT aa.Armazem,aa.Artigo,aa.Lote, aa.StkActual - isnull( (SELECT sum(ln.Quantidade-lns.QuantTrans) FROM CabecDoc cd INNER JOIN LinhasDoc ln ON ln.IdCabecDoc=cd.Id INNER JOIN CabecDocStatus cds ON cds.IdCabecDoc=cd.Id INNER JOIN LinhasDocStatus lns ON lns.IdLinhasDoc=ln.Id WHERE cds.Anulado='0' AND cd.TipoDoc='ECL' AND ln.Armazem=aa.Armazem AND ln.Artigo=aa.Artigo AND ln.Lote=aa.Lote AND lns.Fechado='0' AND lns.EstadoTrans='P' ),0) AS 'StkActual', aa.Localizacao, aa.PCMedio, aa.PCUltimo FROM ArtigoArmazem aa WHERE aa.Armazem='INO' AND aa.Artigo in (SELECT a.Artigo FROM primundifios.dbo.Artigo a WHERE a.Artigo=aa.Artigo) And aa.StkActual > 0 ");

                        Module1.AbreEmpresa("MUNDIFIOS");
                        DocInt = new IntBEDocumentoInterno();

                        DocInt = Module1.emp.Internos.Documentos.Edita("ES", 1, "INO", "000");

                        DocInt.Linhas.RemoveTodos();

                        DocInt.Data = DateTime.Now;

                    }

                    try
                    {
                        for (int j = 1; j <= listaDoc.NumLinhas(); j++)
                        {
                            Module1.emp.Internos.Documentos.AdicionaLinha(DocInt, listaDoc.Valor("Artigo"), listaDoc.Valor("Armazem"), listaDoc.Valor("Localizacao"), listaDoc.Valor("Lote"), Math.Round(Convert.ToDouble(listaDoc.Valor("PCMedio")), 5), 0, listaDoc.Valor("StkActual"), 1, 1, 1);
                            DocInt.Linhas.GetEdita(DocInt.Linhas.NumItens).DataStock = DateTime.Now;
                            listaDoc.Seguinte();
                        }

                        // ----------------------------------
                        // GRAVAÇÃO DO DOCUMENTO
                        Module1.emp.Internos.Documentos.Actualiza(DocInt);
                        // GRAVAÇÃO DO DOCUMENTO
                        // ----------------------------------

                        // MENSAGEM FINAL

                        string strDetalhe;
                        strDetalhe = Constants.vbNullString;

                        strDetalhe = strDetalhe + "Documento Interno: " + DocInt.Tipodoc + " Nº " + System.Convert.ToString(DocInt.NumDoc) + "/" + DocInt.Serie + Constants.vbCrLf;

                        MessageBox.Show("Documento gerado com sucesso. \nInformações: " + strDetalhe, "Sucesso!", MessageBoxButtons.OK, MessageBoxIcon.Information);

                        //'   MENSAGEM FINAL
                        //'----------------------------------

                        DocInt = null;

                        Module1.FechaEmpresa();

                        break;

                    }
                    catch (Exception ex)
                    {
                        MessageBox.Show("Erro ao gerar Documento. \nExceção: " + ex.ToString(), "", MessageBoxButtons.OK, MessageBoxIcon.Error);
                        Module1.FechaEmpresa();
                    }

                }
            }

        }


        string VarFrom;
        string VarTo;
        string VarAssunto;
        string VarTextoInicialMsg;
        string VarMensagem;
        string VarUtilizador;
        string VarLinhas;

        private void EnvioEmailRama()
        {
            int ln;
            VarFrom = "";

            VarTo = "informatica@mundifios.pt;jafernandes@mundifios.pt";

            if (DateTime.Now.TimeOfDay >= new TimeSpan(7, 0, 0) && DateTime.Now.TimeOfDay <= new TimeSpan(12, 59, 0))
                VarTextoInicialMsg = "Bom dia,";
            else if (DateTime.Now.TimeOfDay >= new TimeSpan(13, 0, 0) && DateTime.Now.TimeOfDay <= new TimeSpan(19, 59, 0))
                VarTextoInicialMsg = "Boa tarde,";
            else
                VarTextoInicialMsg = "Boa noite,";

            VarAssunto = "(Rama Inovafil) Encomenda a Fornecedor: " + Strings.Format(this.DocumentoCompra.NumDoc, "####") + "/" + this.DocumentoCompra.Serie;

            VarUtilizador = Aplicacao.Utilizador.Utilizador;

            VarLinhas = "";

            for (ln = 1; ln <= this.DocumentoCompra.Linhas.NumItens; ln++)

                VarLinhas = VarLinhas + "Linha " + ln + ":                         " + this.DocumentoCompra.Linhas.GetEdita(ln).Artigo + " - Armazem:" + this.DocumentoCompra.Linhas.GetEdita(ln).Armazem + " - Lote:" + this.DocumentoCompra.Linhas.GetEdita(ln).Lote + " - Desc:" + this.DocumentoCompra.Linhas.GetEdita(ln).Descricao + " - Quantidade:" + this.DocumentoCompra.Linhas.GetEdita(ln).Quantidade + this.DocumentoCompra.Linhas.GetEdita(ln).Unidade + " - Prec.Unit:" + this.DocumentoCompra.Linhas.GetEdita(ln).PrecUnit + this.DocumentoCompra.Moeda + " - Data Entrega:" + this.DocumentoCompra.Linhas.GetEdita(ln).DataEntrega + Strings.Chr(13) + "";


            VarMensagem = VarTextoInicialMsg + Strings.Chr(13) + Strings.Chr(13) + Strings.Chr(13) + "Foi emitido uma Encomenda a Fornecedor no Primavera:" + Strings.Chr(13) + Strings.Chr(13) + ""
                        + "Empresa:                         " + BSO.Contexto.CodEmp + " - " + BSO.Contexto.IDNome + Strings.Chr(13) + ""
                        + "Utilizador:                      " + VarUtilizador + Strings.Chr(13) + Strings.Chr(13) + ""
                        + "Fornecedor:                      " + this.DocumentoCompra.Entidade + " - " + Strings.Replace(BSO.Base.Fornecedores.Edita(this.DocumentoCompra.Entidade).Nome, "'", "") + Strings.Chr(13) + ""
                        + "Documento:                       " + this.DocumentoCompra.Tipodoc + " " + Strings.Format(this.DocumentoCompra.NumDoc, "#,###") + "/" + this.DocumentoCompra.Serie + Strings.Chr(13) + Strings.Chr(13) + Strings.Chr(13) + ""
                        + VarLinhas + Strings.Chr(13) + ""
                        + "Cumprimentos";




            BSO.DSO.ExecuteSQL("INSERT INTO [PRIEMPRE].[DBO].[MENSAGENSEMAIL]  ([Data], [From], [To], [CC], [BCC], [Assunto], [Mensagem], [Anexos], [Formato], [Utilizador]) VALUES('" + Strings.Format(DateTime.Now, "yyyy-MM-dd HH:mm:ss") + "', '" + VarFrom + "', '" + VarTo + "','','','" + VarAssunto + "','" + VarMensagem + "','',0,'" + VarUtilizador + "' )");
        }


    }
}
